server/lizmap_filter.py

```diff
@@ -0,0 +1,53 @@
+import os
+import json
+
+from qgis.core import Qgis, QgsMessageLog
+from qgis.server import QgsServerInterface, QgsOgcServiceException
+
+from qgis.server import QgsServerFilter
+from qgis.core import Qgis, QgsMessageLog
+
+
+class LizmapFilter(QgsServerFilter):
+
+    def __init__(self, server_iface: 'QgsServerInterface') -> None:
+        QgsMessageLog.logMessage('LizmapFilter.init', 'lizmap', Qgis.Info)
+        super(LizmapFilter, self).__init__(server_iface)
+
+        self.iface = server_iface
+
+    def requestReady(self):
+        config_path = self.iface.configFilePath()
+        if not os.path.exists(config_path):
+            return
+
+        config_path += '.cfg'
+        if not os.path.exists(config_path):
+            return
+
+        cfg = None
+        with open(config_path, 'r') as cfg_file:
+            try:
+                cfg = json.loads(cfg_file.read())
+            except Exception:
+                return
+
+        if not cfg:
+            return
+
+        cfg_options = cfg['options']
+        if 'acl' not in cfg_options or not cfg_options['acl']:
+            return
+
+        cfg_acl = cfg_options['acl']
+
+        handler = self.iface.requestHandler()
+
+        groups = handler.requestHeader('X-Lizmap-User-Groups').split(',')
+        groups = [g.strip() for g in groups]
+        for g in groups:
+            if g in cfg_acl:
+                return
+
+        raise QgsOgcServiceException('Forbidden', 'No ACL permissions', responseCode=403)
```

Comment:
That's will not work, afaik python exception will not be translated to C++ exception. You have to set the return code using the request handler.

Reply:
can I use ?
```
handler.setServiceException(QgsOgcServiceException('Forbidden', 'No ACL permissions', responseCode=403))
handler.setStatusCode(403)
```

Reply:
If you use the first one, you do not need to use `setStastusCode`. If you use `setStatusCode` return a body error message.  

`setServiceException` may ensure that the error message will be properly formated.